flink-tests/src/test/java/org/apache/flink/runtime/jobmaster/JobMasterStopWithSavepointITCase.java

```diff
@@ -384,24 +380,28 @@ protected void processInput(MailboxDefaultAction.Controller controller) throws E
     /** A {@link StreamTask} that simply waits to be terminated normally. */
     public static class NoOpBlockingStreamTask extends NoOpStreamTask {
 
-        private final transient OneShotLatch finishLatch;
+        private transient MailboxDefaultAction.Suspension suspension;
 
         public NoOpBlockingStreamTask(final Environment environment) throws Exception {
             super(environment);
-            this.finishLatch = new OneShotLatch();
         }
 
         @Override
         protected void processInput(MailboxDefaultAction.Controller controller) throws Exception {
             invokeLatch.countDown();
-            finishLatch.await();
-            controller.allActionsCompleted();
+            if (suspension == null) {
+                suspension = controller.suspendDefaultAction();
+            } else {
+                controller.allActionsCompleted();
+            }
```

Comment:
This change removes blocking from `NoOpBlockingStreamTask`, right?
So it's not clear to me what the tests are verifying now.

Probably, the they were written before the mailbox was introduced when different threads were calling `processInput` and `finishTask`. And it tested that they can't block each other.
Now they **can**, so the test seems invalid to me.

WDYT?

Reply:
No. The tests were/are verifying behaviour after triggering stop with savepoint. 

The blocking was supposed to keep the task running until the `finishTask` is called. With my changes it still maintains that expected behaviour. Only when `finishTask` is called we report to the `mailbox` that we won't produce any more records.

Reply:
Makes sense.